apis/pipelines/v1alpha5/provider_types.go

```diff
@@ -0,0 +1,71 @@
+package v1alpha5
+
+import (
+	"github.com/sky-uk/kfp-operator/apis"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ProviderSpec struct {
+	Image         string `json:"image" yaml:"image"`
+	ExecutionMode string `json:"executionMode" yaml:"executionMode"`
+	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
+	ServiceAccount      string                           `json:"serviceAccount" yaml:"serviceAccount"`
+	DefaultBeamArgs     []apis.NamedValue                `json:"defaultBeamArgs,omitempty" yaml:"defaultBeamArgs,omitempty"`
+	PipelineRootStorage string                           `json:"pipelineRootStorage" yaml:"pipelineRootStorage"`
+	Parameters          map[string]*apiextensionsv1.JSON `json:"parameters,omitempty" yaml:"parameters,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:shortName="mlpr"
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="SynchronizationState",type="string",JSONPath=".status.conditions[?(@.type == 'Synchronized')].reason"
+//+kubebuilder:storageversion
+
+type Provider struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ProviderSpec   `json:"spec,omitempty"`
+	Status ProviderStatus `json:"status,omitempty"`
+}
+
+type ProviderStatus struct {
+	Conditions Conditions `json:"conditions,omitempty"`
+}
+
+func (ps *ProviderStatus) SetSynchronizationState(state apis.SynchronizationState, message string, observedGeneration int64) {
```

Comment:
This Code seems not to be used other than in the test? Do we need the `SynchronizationState` for this piece of work at all?

Reply:
removed in [a6ed438](https://github.com/sky-uk/kfp-operator/pull/353/commits/a6ed43815cfa649f714e9eb4e18d6785933c1513)